reward/staking_manager.go

```diff
@@ -92,7 +117,9 @@ func (sm *StakingManager) GetStakingInfo(blockNum uint64) *StakingInfo {
 
 // updateStakingInfo updates staking info in cache and db created from given block number.
 func (sm *StakingManager) updateStakingInfo(blockNum uint64) (*StakingInfo, error) {
-	stakingInfo, err := sm.addressBookConnector.getStakingInfoFromAddressBook(blockNum)
+	stakingBlockNumber := params.CalcStakingBlockNumber(blockNum)
```

Comment:
This can change `GetStakingInfo`'s result.
If `GetStakingInfo` called with block #N, it will return the result of #N - 2*stakingInterval 2*stakingInterval.
This is not good idea.

Reply:
Thank you for pointing it out. Its now modified.